Runtime/Model/Metrics/UniqueEventsSubmissionQueue.cs

```diff
@@ -0,0 +1,45 @@
+﻿using Backtrace.Unity.Common;
+using Backtrace.Unity.Json;
+using Backtrace.Unity.Model.JsonData;
+using System.Collections.Generic;
+
+[assembly: System.Runtime.CompilerServices.InternalsVisibleTo("Backtrace.Unity.Tests.Runtime")]
+namespace Backtrace.Unity.Model.Metrics
+{
+    internal sealed class UniqueEventsSubmissionQueue : MetricsSubmissionQueue<UniqueEvent>
+    {
+        private const string Name = "unique-events";
+        private readonly AttributeProvider _attributeProvider;
+        public UniqueEventsSubmissionQueue(
+            string universeName,
+            string token,
+            IBacktraceHttpClient httpClient,
+            AttributeProvider attributeProvider)
+            : base(Name, universeName, token, httpClient)
+        {
+            _attributeProvider = attributeProvider;
+        }
+
+        public override void StartWithEvent(string eventName)
+        {
+            var uniqueEventAttributes = _attributeProvider.Get();
+            if (uniqueEventAttributes.TryGetValue(eventName, out string value) && !string.IsNullOrEmpty(value))
+            {
+                Events.AddLast(new UniqueEvent(eventName, DateTimeHelper.Timestamp(), uniqueEventAttributes));
+            }
+            Send();
+        }
+
+        internal override IEnumerable<BacktraceJObject> GetEventsPayload(ICollection<UniqueEvent> events)
+        {
+            var uniqueEventsJson = new List<BacktraceJObject>();
+            foreach (var uniqueEvent in events)
+            {
+                uniqueEventsJson.Add(uniqueEvent.ToJson());
+                uniqueEvent.UpdateTimestamp(DateTimeHelper.Timestamp(), _attributeProvider.GenerateAttributes());
+            }
+
```

Comment:
Why don't we need to clear the events here like we do with SummedEvents? I'm sure it has something to do with the complicated spec lol, it will just help my understanding when I do these implementations myself

Reply:
we don't send the same summed events every time (like heartbeat/unique/(you know the right name ( ͡° ͜ʖ ͡°))